alf/algorithms/distributed_off_policy_algorithm.py

```diff
@@ -0,0 +1,541 @@
+        """
+        self._params_lock = threading.Lock()
+        # start the params sending subprocess
+        th = threading.Thread(
+            target=send_params_to_unroller,
+            args=(self._core_alg, self._port, self._params_lock))
+        th.daemon = True
+        th.start()
+
+    def _create_data_receiver_subprocess(self):
+        """Create a proc to receive experience data from the unroller.
+        """
+        # First create the replay buffer in the main process. For this, we need
+        # to create a dummy experience to set up the replay buffer.
+        time_step = self._env.current_time_step()
+        rollout_state = self.get_initial_rollout_state(self._env.batch_size)
+        alg_step = self.rollout_step(time_step, rollout_state)
+        exp = make_experience(time_step, alg_step, rollout_state)
+        exp = alf.utils.common.prune_exp_replay_state(
+            exp, self._use_rollout_state, self.rollout_state_spec,
+            self.train_state_spec)
+        alf.config('ReplayBuffer', allow_multiprocess=True)
+        self._set_replay_buffer(exp)
+
+        # Create a shared value to record the total number of experience samples
+        # received from the unroller
+        self._total_exps_received = mp.Value('i', 0)
+
+        # In the case of DDP, each subprocess is spawned. By default, if we create
+        # a new subprocess, the default start method inherited is spawn. In this case,
+        # we need to explicitly set the start method to fork, so that the daemon
+        # subprocess can share torch modules.
+        mp.set_start_method('fork', force=True)
+        # start the data receiver subprocess
+        process = mp.Process(
+            target=receive_experience_data,
+            args=(self._replay_buffer, max(0,
+                                           PerProcessContext().ddp_rank),
+                  self._unroller_ip, self._unroller_port,
+                  self._total_exps_received),
+            daemon=True)
+        process.start()
+
+    def _send_trainer_info_to_unroller(self):
+        """Create a REP socket and send the number of workers to the unroller,
+        so that the unroller is able to know the worker ids
+
+        'worker-0', 'worker-1', ...
+
+        to route experience data to.
+
+        Also send the trainer's ip and port to the unroller.
+        """
+        socket, cxt = create_zmq_socket(zmq.REQ, self._unroller_ip,
+                                        self._unroller_port)
+        logging.info(
+            '[worker-0] Waiting to send the unroller the number of workers...')
+        trainer_ip = get_local_ip()
+        socket.send_string(
+            f'worker-0: {PerProcessContext().num_processes}, {trainer_ip}, {self._port}'
+        )
+        message = socket.recv_string()
+        assert message == 'unroller: ok'
+        socket.close()
+        cxt.term()
+
+    @property
+    def utd(self):
+        with self._total_exps_received.get_lock():
```

Comment:
the total number of experiences can be obtained from `replay_buffer.get_current_position().sum()`